jobs/build.Jenkinsfile

```diff
@@ -516,3 +457,62 @@ lock(resource: "build-${params.STREAM}") {
         """)
     }
 }}}} // finally, cosaPod, timeout, and locks finish here
+
+// This does an early archive of just the OSTree. This has the desired side
+// effect of reserving our build ID before we fork off multi-arch builds.
+def archive_ostree(version, basearch, s3_stream_dir) {
+    stage('Archive OSTree') {
+        def acl = pipecfg.s3.acl ?: 'public-read'
+        // run with --force here in case the previous run of the
+        // pipeline died in between buildupload and bump_builds_json()
+        pipeutils.shwrapWithAWSBuildUploadCredentials("""
+        cosa buildupload --force --skip-builds-json --artifact=ostree \
+            s3 --aws-config-file=\${AWS_BUILD_UPLOAD_CONFIG} \
+            --acl=${acl} ${s3_stream_dir}/builds
+        """)
+        pipeutils.bump_builds_json(
+            params.STREAM,
+            version,
+            basearch,
+            s3_stream_dir,
+            acl)
+    }
+}
+
+def run_multiarch_jobs(arches, src_commit, version, cosa_img) {
+    stage('Fork Multi-Arch Builds') {
+        for (arch in arches) {
+            // We pass in FORCE=true here since if we got this far we know
+            // we want to do a build even if the code tells us that there
+            // are no apparent changes since the previous commit.
+            build job: 'build-arch', wait: false, parameters: [
+                booleanParam(name: 'FORCE', value: true),
+                booleanParam(name: 'ALLOW_KOLA_UPGRADE_FAILURE', value: params.ALLOW_KOLA_UPGRADE_FAILURE),
+                string(name: 'SRC_CONFIG_COMMIT', value: src_commit),
+                string(name: 'COREOS_ASSEMBLER_IMAGE', value: cosa_img),
+                string(name: 'STREAM', value: params.STREAM),
+                string(name: 'VERSION', value: version),
+                string(name: 'ARCH', value: arch),
+                string(name: 'PIPECFG_HOTFIX_REPO', value: params.PIPECFG_HOTFIX_REPO),
+                string(name: 'PIPECFG_HOTFIX_REF', value: params.PIPECFG_HOTFIX_REF)
+            ]
+        }
+    }
+}
+
+def run_release_job(buildID) {
+    stage('Publish') {
+        // Since we are only running this stage for non-production (i.e. mechanical
+        // and development) builds we'll default to not doing AWS AMI replication.
+        // We'll also default to allowing failures for additional architectures.
```

Comment:
We should update this comment. Suggestion:

```
// Since we are only running this stage for non-production (i.e.
// mechanical and development) builds we'll default to allowing
// failures for additional architectures.
```